src/main/java/com/berkinozturk/event/service/AuthenticationService.java

```diff
@@ -15,32 +17,34 @@
 @RequiredArgsConstructor
 public class AuthenticationService {
 
-
     private final UserRepository userRepository;
     private final JwtService jwtService;
     private final AuthenticationManager authenticationManager;
+    private final UserMapper userMapper;
 
     public AuthenticationResponse register(RegisterRequest request) {
-        var user = UserEntity.builder()
-                .username(request.getUsername())
-                .password(request.getPassword())
-                .email(request.getEmail())
-                .fullName(request.getFullName())
-                .role(request.getRole())
-                .build();
+        UserEntity user = userMapper.toUserEntity(request);
         userRepository.save(user);
-
-        var jwtToken = jwtService.generateToken(user);
-        return AuthenticationResponse.builder()
-                .token(jwtToken)
-                .build();
-
+        try {
+            String jwtToken = jwtService.generateToken(user);
+            return AuthenticationResponse.builder()
+                    .token(jwtToken)
+                    .build();
+        } catch (Exception ex) {
+            throw new RuntimeException("Token generation failed: " + ex.getMessage());
+        }
     }
 
     public AuthenticationResponse authenticate(AuthenticationRequest request) {
-        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword()));
-        var user = userRepository.findByEmail(request.getEmail()).orElseThrow();
-        var jwtToken = jwtService.generateToken(user);
+        // The problem with auth is here, bad credentials
+        try {
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getEmail(), request.getPassword()));
+        } catch (Exception ex) {
+            throw new RuntimeException("Authentication failed: " + ex.getMessage());
+        }
```

Comment:
When you use try catch, you have to handle only the errors you can. For example, a lot of things can go bad there and you will throw AuthenticationFailed. It could be a bug on your end and the users will get AuthenticationFailed, as a user you'd feel frustrated because, you know the credentials are correct.

Handle the AuthenticationException only then, you can inform them Authentication is failing.

For the rest of the exceptions, you can just return 500 - Internal Server Error.